### ISSUE TITLE
How to create multiple CSV output files?

### ISSUE THREAD

**Issue by pmackay** [nevern02/rodimus#3]
Hi, I'm really interested to try out Rodimus, but my use case is one where I need to output multiple CSV files. Can you suggest if this is a good fit for Rodimus, and what the best approach might be?

**Comment by nevern02** [comment]
One of planned features we've done some experimenting on is sending the output from a single step to multiple steps.  There's actually a branch in the repo called "multi_step_output" that has some of that work.  You're welcome to take a look or make some tweaks/suggestions.  I'm not totally sold on the approach there, but it's something to think about.

Otherwise, the more straightforward approach would be to split your work into multiple transformations, where each one represents a stage in the process.  Transformation 1 would take your source data, transform it, and output it into some kind of staging area/database/file.  Then, write 2, 3, or however many other transformations, each of which reads from the staging area and writes into its own CSV.  In that scenario, Transformation 1 would need to run before anything else, but then the other transformations could all run in parallel.  I typically find that if I'm having trouble coming up with a solution to some ETL operation I need to perform, then the answer is to split it into smaller steps and/or multiple transformations.

**Comment by pmackay** [comment]
Can I clarify, if the last step is the one where the outgoing pipe is customised to write to a CSV (or stdout or whatever), how would it then pass data onto another step which might be writing a different CSV, if each step is conditional on the output of the step before?

**Comment by nevern02** [comment]
You're right.  Since the step's outgoing pipe is writing to a CSV file, it can't then pass the data on to another step.  Writing to multiple output steps is still an experiment/idea I'm playing with.

My suggested solution is to use multiple transformations, not additional steps in the same transformation.    Transformation A writes to CSV File A, Transformation B writes to CSV File B, etc..  Before either of those transformations run, you would have another transformation who's job is essentially to prepare or "stage" the data for your CSV transformations.
